### DISCUSSION TITLE
banded ridge regression

### DISCUSSION THREAD

**Discussion by alireza-kr** [cvnlab/GLMsingle#139] [Ideas]
Hello

I am curious to know your thoughts on using **banded ridge regression** (La Tour et al., 2022) to disentangle correlated response estimates between trials that are close in time.

- La Tour, T. D., Eickenberg, M., Nunez-Elizalde, A. O., & Gallant, J. L. (2022). Feature-space selection with banded ridge regression. NeuroImage, 264, 119728. https://doi.org/10.1016/j.neuroimage.2022.119728

**Comment by kendrickkay** [comment]
In GLMsingle, there is only one set of predictors that sort of "group" together, namely, the single trial predictors.  (The other predictors, nuisance/drift/etc., are fitted using least squares up front.)  So, banded ridge regression doesn't seem to apply in any straightforward way...  did you have something in mind?

**Comment by alireza-kr** [comment]
Thank you Kendrick for your answer.

I read in your paper [1] about the instabilities in GLM estimation that can arise from having correlated single-trial predictors. You suggested using ridge regression to address this. However, ridge regression applies the same regularization hyperparameter to all features. What I suggest is using banded ridge regression, which allows for a separate regularization level for each feature.

Prince, J. S., Charest, I., Kurzawski, J. W., Pyles, J. A., Tarr, M. J., & Kay, K. N. (2022). Improving the accuracy of single-trial fMRI response estimates using GLMsingle. eLife, 11. https://doi.org/10.7554/elife.77599

**Comment by kendrickkay** [comment]
Yes, but are you suggesting a different regularization level for every single trial?  That seems a bit crazy
